plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergPlugin.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.testing.TestingConnectorContext;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Map;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.nio.file.Files.createTempDirectory;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestIcebergPlugin
+{
+    private Path tempDirectory;
+
+    @BeforeClass
+    public void setup()
+            throws IOException
+    {
+        tempDirectory = createTempDirectory(getClass().getSimpleName());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        deleteRecursively(tempDirectory, ALLOW_INSECURE);
+    }
+
+    @Test
+    public void testCreateConnector()
+    {
+        ConnectorFactory factory = getConnectorFactory();
+        // simplest possible configuration
+        factory.create("test", Map.of("hive.metastore.uri", "thrift://foo:1234"), new TestingConnectorContext()).shutdown();
+    }
+
+    @Test
+    public void testThriftMetastore()
+    {
+        ConnectorFactory factory = getConnectorFactory();
+
+        factory.create(
+                "test",
+                Map.of(
+                        "hive.metastore", "thrift",
+                        "hive.metastore.uri", "thrift://foo:1234"),
+                new TestingConnectorContext())
+                .shutdown();
+    }
+
+    @Test
+    public void testGlueMetastore()
+    {
+        ConnectorFactory factory = getConnectorFactory();
+
+        assertThatThrownBy(() -> factory.create(
+                "test",
+                Map.of("hive.metastore", "glue"),
+                new TestingConnectorContext()))
+                .hasMessageContaining("Explicit bindings are required and HiveMetastore is not explicitly bound");
```

Comment:
The error messages thrown here are not very intuitive to the users. I think a more explicit message like `Glue is not supported a Hive metastore type in Iceberg connector` is better. And ideally we should throw this message for all configurations that tries to set `hive.metastore=glue`, instead of showing different messages.

I don't know if it is too hard to achieve that in the dependency injection framework. If not, we should at least make this clear in the documentation.

Reply:
> The error messages thrown here are not very intuitive to the users. 


i do agree with that. Same applies to Hive connector and was a deliberate choice there, so didn't want to change that.